jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const logger = require('../common/logger');
+const BaseJob = require('./BaseJob');
+const catalog = require('../common/models/catalog');
+const config = require('../common/config');
+const utils = require('../common/utils');
+const maas = require('../data-access-layer/metering');
+
+const apiServerClient = require('../data-access-layer/eventmesh').apiServerClient;
+const CONST = require('../common/constants');
+
+class MeterInstanceJob extends BaseJob {
+  static run(job, done) {
+    return Promise.try(() => {
+      logger.debug('Starting MeterInstanceJob Job');
+      this.getInstanceEvents()
+        .tap(events => logger.info('recieved events -> ', events))
+        .then(events => this.meter(events))
+        .then((meterResponse) => this.runSucceeded(meterResponse, job, done))
+        .catch((err) => this.runFailed(err, {}, job, done));
+    });
+  }
+
+  static getInstanceEvents() {
+    const options = {
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.INSTANCE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.SFEVENT,
+      query: {
+        labelSelector: `meter_state in (${CONST.METER_STATE.TO_BE_METERED},${CONST.METER_STATE.FAILED})`
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  static meter(events) {
+    logger.info(`Number of events to be metered in this run - ${events.length}`);
+    return Promise.try(() => {
+      let successCount = 0,
+        failureCount = 0,
+        failedEvents = [];
+      return Promise.map(events, (event) => {
+          return this
+            .sendEvent(event)
+            .then((status) => {
+              if (status) {
+                successCount++;
+              } else {
+                failureCount++;
+                failedEvents.push(event);
+              }
+            })
+            .catch(err => {
+              logger.error(`Error occurred while metering event : `, err);
+            });
+        }).then(() => {
+          return {
```

Comment:
unfortunately it doesn't e.g:
```
function logger(callback) {
  console.log(callback())
}

logger(() => { return { key: "value", key1: "value1" }});
// logger(() => { key: "value", key1: "value1" } ); // throws syntax error
```

Reply:
just to add, if you want to avoid a `return`, you can use the new ES6 way for functional composition:

e.g.

```
.then(() => ({key1: 'value1', key2: 'value2'})
```